src/tests/Snackbar.test.tsx

```diff
@@ -0,0 +1,81 @@
+import { waitForElementToBeRemoved } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { act } from 'react-dom/test-utils';
+
+import SearchView from '../components/Search/SearchView';
+import getTestData from './utils/fixtures';
+import { renderWithRouter, screen } from './utils/test-utils';
+
+describe('Snackbar', () => {
+  it('should dismiss an alert when close button is clicked', async () => {
+    const { testData } = getTestData();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => ({
+          results: testData,
+        }),
+      }),
+    ) as jest.Mock;
+    // set delay to null to prevent test time-out due to useFakeTimers
+    const user = userEvent.setup({ delay: null });
+
+    renderWithRouter(<SearchView />);
+
+    // focus input to show results
+    const searchInput = screen.getByRole('textbox');
+    await user.click(searchInput);
+
+    await user.click(screen.getByTestId('checkbox-0'));
+    await user.click(screen.getByTestId('checkbox-1'));
+    await user.click(screen.getByTestId('checkbox-2'));
+    await user.click(screen.getByTestId('checkbox-3'));
+    await user.click(screen.getByTestId('checkbox-4'));
+
+    const alert = screen.getByText('Maximum 4 Revisions');
+
+    const closeButton = screen.getByTestId('alert-close');
+    await user.click(closeButton);
+
+    await waitForElementToBeRemoved(closeButton);
+    expect(alert).not.toBeInTheDocument();
+  });
+
+  it('should dismiss an alert after 6 seconds', async () => {
+    const { testData } = getTestData();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => ({
+          results: testData,
+        }),
+      }),
+    ) as jest.Mock;
+    jest.spyOn(global, 'setTimeout');
+    // set delay to null to prevent test time-out due to useFakeTimers
+    const user = userEvent.setup({ delay: null });
+
+    renderWithRouter(<SearchView />);
+
+    // focus input to show results
+    const searchInput = screen.getByRole('textbox');
+    await user.click(searchInput);
+
+    await user.click(screen.getByTestId('checkbox-0'));
+    await user.click(screen.getByTestId('checkbox-1'));
+    await user.click(screen.getByTestId('checkbox-2'));
+    await user.click(screen.getByTestId('checkbox-3'));
+    await user.click(screen.getByTestId('checkbox-4'));
+
+    const alert = screen.getByText('Maximum 4 Revisions');
+
+    const closeButton = screen.getByTestId('alert-close');
+    await user.click(closeButton);
```

Comment:
It seems to me that if you want to test dismissing the alert after 6 seconds then this line is not needed. Am I wrong?

Reply:
ah--you are right. thanks for catching this!